src/components/templates/AddStory.js

```diff
@@ -23,13 +23,24 @@ const AddStory = ({ toggleModal }) => {
       !newStory.title.trim()
       || !newStory.content.trim()
     ) return;
-
+    //don't attempt if the user is not logged in
+    if (
+      !currentUser
+    ) return;
+    //don't attempt if there is no local user date
+    if (
+      !Date.now()
+    ) return;
     try {
       await db.collection('links')
         .add({
           title: newStory.title,
           content: newStory.content,
           author: currentUser.displayName,
+          authorId: currentUser.uid,
+          timestamp: Date.now(),
```

Comment:
First, let's change this to `createdDate`

```suggestion
          createdDate: Date.now(),
```

Secondly, let's use Firebase's Timestamp class for this. The reason being, Firebase's Timestamp records the time when the document is added to the collection, whereas Date.now() gets the time of when that Date.now() code was fired. 

We don't actually want the time the user hits submit, but rather the time that the document was finally added to the collection.

There's a link on the Issue that shows you how to use Firebase.Timestamp